src/main/java/org/hisp/dhis/query/Query.java

```diff
@@ -18,6 +18,8 @@
 
     private boolean expandAssociations = false;
 
+    private boolean rootJunctionOr = false;
```

Comment:
I think I would prefer an enum `RootJunction` with `AND` and `OR` here, just to model it nicely and, though unlikely, allow for future extension.

Reply:
@larshelge I have made the change.